scvi/data/fields/_protein.py

```diff
@@ -73,29 +74,30 @@ def _get_batch_mask_protein_data(self, adata: AnnData) -> Optional[dict]:
         The mask is over cell measurement columns that are present (observed)
         in each batch. Absence is defined by all 0 for that protein in that batch.
         """
-        pro_exp = self.get_field_data(adata)
-        pro_exp = pro_exp.to_numpy() if isinstance(pro_exp, pd.DataFrame) else pro_exp
-        batches = adata.obs[self.batch_key].values
-        batch_mask = {}
-        for b in np.unique(batches):
-            b_inds = np.where(batches.ravel() == b)[0]
-            batch_sum = pro_exp[b_inds, :].sum(axis=0)
-            all_zero = batch_sum == 0
-            batch_mask[b] = ~all_zero
-
-        if np.sum([~b[1] for b in batch_mask.items()]) > 0:
-            logger.info("Found batches with missing protein expression")
-            return batch_mask
-
+        if self.use_batch_mask:
+            pro_exp = self.get_field_data(adata)
+            pro_exp = (
+                pro_exp.to_numpy() if isinstance(pro_exp, pd.DataFrame) else pro_exp
+            )
+            batches = self.batch_field.get_field_data(adata)
```

Comment:
does this field need to be registered before?

Reply:
yes order matters, they get run in the same order they are passed in